src/test/resources/eager/reconstructs-words-from-inside-nested-expressions.expected.jinja

```diff
@@ -0,0 +1,2 @@
+{% set foo = 'a' %}{% do deferred.append('{{ foo }}') %}
```

Comment:
After rendering should this become
```
{% do deferred.append('a') %}
{{ deferred[0] }}
```

Reply:
It would not, because if you were to do:
```
{% set my_list = [] %}
{% do my_list.append('{{ 1 + 2 }}') %}
Printed: {% print my_list[0] %}
Expressioned: {{ my_list[0] }}
```
The output is:
```
Printed: {{ 1 +  2 }}
Expressioned: 3
```

It only does nested interpretation if it's used inside of an expression node. So we shouldn't evaluate `{{ foo }}` yet.

I can make this more clear by updating the unit test/adding another one that demonstrates the second pass

Reply:
This is a bit confusing now. What do we expect this?
```
{% set foo = 'hello' %}
Printed: {% print foo %}
Expressioned: {{ foo }}
```

Reply:
That will be:
```
Printed: hello
Expressioned: hello
```

Reply:
It is confusing that jinjava supports nested interpretation, but only inside expression nodes... But that is how it was designed

Reply:
OK. So only expression nested in `{{ }}` will be interpreted. Thanks.